ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -197,7 +197,32 @@ private[backend] object H2StorageBackend
       connectionInitHook: Option[Connection => Unit],
   )(implicit loggingContext: LoggingContext): DataSource = {
     val h2DataSource = new org.h2.jdbcx.JdbcDataSource()
-    h2DataSource.setUrl(jdbcUrl)
+
+    // H2 (org.h2.jdbcx.JdbcDataSource) does not support setting the user/password within the jdbcUrl, so remove
+    // those properties from the url if present and set them separately. This helps run canton tests which set
+    // user/password. Note that Postgres and Oracle support user/password in the URLs, so we don't bother exposing
+    // user/password configs separately from the url just for h2 which is anyway not supported for production.
+    def setKeyValueAndRemoveFromUrl(url: String, key: String, setter: String => Unit): String = {
```

Comment:
If you extract this to a public method, we can unit test this. I'm currently working on adding tests on `StorageBackend` level, where it would be easy to add a test for this.

Reply:
Makes sense. Done.